### PR TITLE
Feature/issue 15749

### PR DESCRIPTION
**Pull request by evonzee**
### What does this PR do?

Adds support for pkg.group_list, pkg.group_info, and pkg.group_diff to modules.pacman.  Includes unit tests.
### What issues does this PR fix or reference?

Issue #15749
### Previous Behavior

Groups (when configured for pkg.installed) would always reinstall, no matter the current state, when running state.pkg.installed.
### New Behavior

Introduces a new toolset, similar to yumpkg, allowing Arch users to use pkg.group_list, pkg.group_info, and pkg.group_diff, and states.pkg.group_installed.
### Tests written?

Yes

### CODE REVIEW

**Comment by cachedout** [comment]
@gtmanfred and @terminalmage would either of you like to jump in here?

File: salt/modules/pacman.py [hunk 1]
```diff
@@ -236,6 +236,158 @@ def list_pkgs(versions_as_list=False, **kwargs):
     return ret
 
 
+def group_list():
+    '''
+    .. versionadded:: ?
```
Comment (terminalmage) [inline salt/modules/pacman.py]: All of the versionadded/versionchanged directives here should have ``?`` replaced with ``Carbon``, the codename for the next feature release.

Comment (evonzee) [inline salt/modules/pacman.py]: Carbonized and PR amended.

**Comment by evonzee** [comment]
Hm, looks like the build failed due to a timeout, perhaps due to load on the build server or network issues?  How can I trigger another build (without pushing junk commits?)

**Comment by cachedout** [comment]
@evonzee I'm fine with merging it without the Ubuntu platform test running, since they won't hit this code anyhow.

**Pull request merged**